Change: Add slightly pitched sounds for Air weapons of Quad Cannon, Minigunner, Gattling Tank, Gattling Turret, Laser Turret

xezon opened the pull request:
* Closes #1331

This change adds slightly pitched sounds to all non rocket Anti Air weapons, including Quad Cannon, Minigunner, Gattling Tank, Gattling Turret, Laser Turret. The default weapons against ground use legacy sound settings.

| Weapon                             | PitchShift Min | PitchShift Max |
|------------------------------------|---------------:|---------------:|
| QuadCannonWeapon                   | -10            | 10             |
| QuadCannonWeaponAir (this)         | -25            | -5             |
| GattlingCannonWeapon               |  0             | 0              |
| GattlingCannonWeaponAir (this)     | -8             | -8             |
| GattlingTankWeapon                 |  0             | 0              |
| GattlingTankWeaponAir (this)       | -8             | -8             |
| RedGuardMinigunnerWeapon           | 0              | 0              |
| RedGuardMinigunnerWeaponAir (this) | -10            | -10            |
| LaserCrusaderWeapon                | -5             | 5              |
| LaserCrusaderWeaponAir (this)      | -15            | -5             |
| LaserCrusaderWeaponAssist (this)   | -10            | 0              |

Adding pitched sounds for air rocket weapons in principle works and sounds good, however adding this to some of the Rocket Soldiers is not without issues. Therefore it is only added to non rocket Anti Air weapons.

The motivation of this change is the express fondness of the former Gattling Tank sound for the Minigunner Air weapon.

The pitch values may need some tweaking.

### Demo

https://user-images.githubusercontent.com/4720891/194398414-877fc599-71ae-45f7-a913-82b8e443aba3.mp4

xezon commented:
This change is now fully functional.

xezon commented:
Anti Pro:
> I like the idea of this approach - if I'm super honest I'd like it to be a little more distinctive across the board if possible?

commy2 commented on Patch104pZH/GameFilesEdited/Data/INI/Weapon.ini:
```diff
@@ -6661,7 +6661,7 @@ Weapon Lazr_PatriotMissileWeaponAir
   LaserName               = Lazr_PatriotLaserBeam
   LaserBoneName           = WEAPONA01
-  FireFX                  = Lazr_WeaponFX_LaserCrusader
+  FireFX                  = Lazr_WeaponFX_LaserCrusader_Air
```
Just use FireSound?

commy2 replied:
Name is weird. I realize it is because of Laser turret recycling Laser tank sounds, but appending "air" to it is just strange.

xezon replied:
I did this this way because the original weapon also spawns the sound this way and therefore it is consistent and a map creator can use it the legacy way.

commy2 commented on Patch104pZH/GameFilesEdited/Data/INI/SoundEffects.ini:
```diff
@@ -3393,6 +3502,20 @@ AudioEvent PatriotBatteryWeapon
+; Patch104p @feature new sound for air weapon variant
+; Not used, because adding rocket air weapons to all units is not easily feasible.
+; For example Demo Tunnel Defender cannot share weapon reload with Suicide weapon.
+AudioEvent PatriotBatteryWeaponAir
```
obsolete

xezon replied:
removed

commy2 commented on Patch104pZH/GameFilesEdited/Data/INI/SoundEffects.ini:
```diff
@@ -2822,6 +2891,20 @@ AudioEvent TankHunterWeapon
+; Patch104p @feature new sound for air weapon variant
+; Not used, because adding rocket air weapons to all units is not easily feasible.
+; For example Demo Tunnel Defender cannot share weapon reload with Suicide weapon.
+AudioEvent TankHunterWeaponAir
```
obsolete

xezon replied:
removed

commy2 commented on Patch104pZH/GameFilesEdited/Data/INI/SoundEffects.ini:
```diff
@@ -2811,6 +2866,20 @@ AudioEvent MissileDefenderWeapon
+; Patch104p @feature new sound for air weapon variant
+; Not used, because adding rocket air weapons to all units is not easily feasible.
+; For example Demo Tunnel Defender cannot share weapon reload with Suicide weapon.
+AudioEvent MissileDefenderWeaponAir
```
obsolete

xezon replied:
removed

commy2 commented on Patch104pZH/GameFilesEdited/Data/INI/SoundEffects.ini:
```diff
@@ -2800,6 +2841,20 @@ AudioEvent RPGTrooperWeapon
+; Patch104p @feature new sound for air weapon variant
+; Not used, because adding rocket air weapons to all units is not easily feasible.
+; For example Demo Tunnel Defender cannot share weapon reload with Suicide weapon.
+AudioEvent RPGTrooperWeaponAir
```
obsolete

xezon replied:
I wanted to keep them to partly act as documentation, and maybe for someone to use them if they like, e.g. Mods.

commy2 replied:
I don't think it is necessary to provide these as in code documentation. It clutters with unused types. Why not put it on Github instead? map.ini can define AudioEvent afaik.

xezon replied:
I will remove them then.

commy2 commented:
This won't work for Laser turret (assist weapon can be used for both ground & air), Demo RPG (breaks Suicide ability), Missile Defender (Laser lock can be used for both ground & air, probably breaks hard-coded logic).

xezon commented:
I removed all previously added unused Air sounds, fixed wrong sound on GattlingCannonWeaponAir, and retweaked pitch on GattlingCannonWeaponAir, GattlingTankWeaponAir, RedGuardMinigunnerWeaponAir to sound more distinct.

From my side this is all good now.

https://user-images.githubusercontent.com/4720891/198822531-fcf8d202-b018-43f6-a134-81c97b0e0e20.mp4

xezon commented:
> This won't work for Laser turret (assist weapon can be used for both ground & air)

I have now added a third audio event variant for Assisted Laser, which sits between Regular and Air pitch levels.

xezon commented:
Is this good to go?

MTKing4 commented:
good to me

xezon commented:
Good.
